scripts/smoke_tests/setup.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+import sys
+import json
+import time
+import threading
+
+import requests
+
+from tests.functional.backend.common import BaseFunctionalTestCase
+
+# Amount to reduce chance of collision where multiple test instances select the same collection to test against
+NUM_TEST_COLLECTIONS = 10
+TEST_ACCT_CONTACT_NAME = "Smoke Test User"
+
+
+class SmokeTestsInitializer(BaseFunctionalTestCase):
+
+    def __init__(self):
+        super().setUpClass()
+        super().setUp()
+        self.headers = {"Cookie": f"cxguser={self.curator_cookie}", "Content-Type": "application/json"}
+
+    def get_collection_count(self):
+        res = self.session.get(f"{self.api}/curation/v1/collections?visiblity=PUBLIC", headers=self.headers)
+        res.raise_for_status()
+        data = json.loads(res.content)
+        num_collections = 0
+        for collection in data['collections']:
+            if collection["contact_name"] == TEST_ACCT_CONTACT_NAME:
+                num_collections += 1
+            if num_collections == NUM_TEST_COLLECTIONS:
+                return num_collections
+        return num_collections
+
+    def create_and_publish_collection(self, dropbox_url):
+        collection_id = self.create_collection()
+        self.upload_and_wait(collection_id, dropbox_url)
+        self.publish_collection(collection_id)
+        print(f"created and published collection {collection_id}")
+
+    def create_collection(self):
+        data = {
+            "contact_email": "example@gmail.com",
+            "contact_name": TEST_ACCT_CONTACT_NAME,
+            "description": "Well here are some words",
+            "links": [{"link_name": "a link to somewhere", "link_type": "PROTOCOL", "link_url": "http://protocol.com"}],
+            "name": "test collection",
+        }
+
+        res = self.session.post(f"{self.api}/dp/v1/collections", data=json.dumps(data), headers=self.headers)
+        res.raise_for_status()
+        data = json.loads(res.content)
+        return data["collection_id"]
+
+    def publish_collection(self, collection_id):
+        body = {"data_submission_policy_version": "1.0"}
+        res = self.session.post(
+            f"{self.api}/dp/v1/collections/{collection_id}/publish", headers=self.headers, data=json.dumps(body)
+        )
+        res.raise_for_status()
+
+    # override
+    def upload_and_wait(self, collection_id, dropbox_url):
```

Comment:
Out of curiosity, why do we need to override?

Reply:
The parent class includes a cleanup 'DELETE {api_url}/v1/datasets' after the dataset is finished uploading

Reply:
Makes sense. Maybe we can add a `delete=True` parameter and pass False here? Only if it's not too complicated though.